File: src/main/java/com/kaiosantiago/laudopro/entity/base/BaseInstrument.java
package com.kaiosantiago.laudopro.entity.base;

import com.kaiosantiago.laudopro.entity.User;
import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;

@EqualsAndHashCode(callSuper = true)
@MappedSuperclass
@Data
public abstract class BaseInstrument extends BaseEntity {
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(length = 100)
    private String description;

    @Column(nullable = false, unique = true)
    private String tag;

    @Column(nullable = false)
    private String manufacturer;

    @Column private String serialNumber;

    @Column private String model;

    @Column private String size;

    @Column private String actuationRange;
}

File: LaudoPro.Domain/Models/BaseInstrument.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace LaudoPro.Domain.Models.Base
{
    public class BaseInstrument : BaseModel
    {
        [Required]
        [MaxLength(100)]
        [Column("description")]
        public string Description { get; set; }

        [Required]
        [Column("tag")]
        [MaxLength(50)]
        public string Tag { get; set; }

        [Required]
        [Column("manufacturer")]
        [MaxLength(100)]
        public string Manufacturer { get; set; }

        [Required]
        [MaxLength(100)]
        [Column("serial_number")]
        public string SerialNumber { get; set; }

        [Required]
        [MaxLength(100)]
        [Column("model")]
        public string Model { get; set; }

        [Required]
        [MaxLength(20)]
        [Column("size")]
        public string Size { get; set; }
    }
}
